test/with_dummyserver/test_socketlevel.py

```diff
@@ -1076,3 +1076,49 @@ def socket_handler(listener):
         self.assertEqual([b'hello, world'], list(r.stream(None)))
 
         done_event.set()
+
+class TestBadContentLength(SocketDummyServerTestCase):
+    def test_strict_content_length(self):
+        done_event = Event()
+
+        def socket_handler(listener):
+            sock = listener.accept()[0]
+
+            buf = b''
+            while not buf.endswith(b'\r\n\r\n'):
+                buf += sock.recv(65536)
+
+            sock.send(
+                b'HTTP/1.1 200 OK\r\n'
+                b'Content-Length: 22\r\n'
+                b'Content-type: text/plain\r\n'
+                b'\r\n'
+                b'hello, world'
+            )
+            done_event.wait(1)
+            sock.close()
+
+        def _make_connection(self):
+            self._start_server(socket_handler)
+            return HTTPConnectionPool(self.host, self.port, maxsize=1)
+
+        def test_get(self):
+            # Test stream read when content length less than headers claim
+            conn = _make_connection(self)
+            get_response = conn.request('GET', url='/', preload_content=False,
+                                        strict_content_length=True)
+            with self.assertRaises(ProtocolError):
```

Comment:
This is currently failing for python 2.6, but made me realize that we aren't catching streams where amt > resp.length. Working on a fix.